app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -94,23 +100,43 @@ public open class MainPresenterImpl(val mapzenLocation: MapzenLocation, val bus:
     }
 
     override fun onRestoreViewState() {
-        if (destination != null) {
-            if(routingEnabled) {
-                resumeRoutingMode()
-            } else {
-                generateRoutePreview()
-            }
-        } else {
-            if (searchResults != null) {
-                mainViewController?.showSearchResults(searchResults?.features)
-            }
+        when (vsm.viewState) {
+            DEFAULT -> onRestoreViewStateDefault()
+            SEARCH -> onRestoreViewStateSearch()
+            SEARCH_RESULTS -> onRestoreViewStateSearchResults()
+            ROUTE_PREVIEW -> onRestoreViewStateRoutePreview()
+            ROUTING -> onRestoreViewStateRouting()
+            ROUTE_DIRECTION_LIST -> onRestoreViewStateRouteDirectionList()
         }
+   }
+
+    private fun onRestoreViewStateDefault() {
+        // Do nothing.
```

Comment:
I was thinking of calling initFindMeButton here!!... But while I am writing this, I do not think its a good idea, since that will always bring you back to your current location instead of restoring the previous last known state. :D.